src/controllers/csv-processor.ts

```diff
@@ -155,6 +157,20 @@ export async function extractTableInformation(fileBuffer: Buffer, fileType: File
     });
 }
 
+function convertBufferToUTF8(buffer: Buffer): Buffer {
+    const fileEncoding = detectCharacterEncoding(buffer)?.encoding;
+    if (!fileEncoding) {
+        logger.warn('Could not detect file encoding for the file');
+        throw new Error('errors.csv.invalid');
+    }
+    if (fileEncoding !== 'UTF-8') {
+        logger.warn(`File is not UTF-8 encoded... File appears to be ${fileEncoding}... Going to try to recode it`);
+        const decodedString = iconv.decode(buffer, fileEncoding);
```

Comment:
Am I right in thinking this is storing the entire CSV data in both `buffer` and `decodedString`?

We might want to look at a streaming solution post-MVP as if they're both several hundred MB we might start seeing memory issues on the deployed containers.

Reply:
Good point... Yes it is.... We need to dump the original buffer once the conversation is done